portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,33 +50,57 @@ private String convertToJsonUsingGson(List<String> messages) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get comments in datastore, by most recent order at the top.
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    Query query = new Query("Comment").addSort(COMMENT_DATE, SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    // Iterate over all entities, get comment text.
-    List<String> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      String commentText = (String) entity.getProperty("text");
-      comments.add(commentText);
-    }
+    // Extract limit on number of comments from query string (default is 5).
+    int limit = (request.getParameter("limit") == null) ?
+      5 : Integer.parseInt(request.getParameter("limit"));
+
+    // Iterate over all entities, get comment.
+    List<Comment> comments = getComments(results, limit);
 
     // Return comments in JSON format.
     response.setContentType("application/json;");
-    String json = convertToJsonUsingGson(comments);
+    String json = convertToJson(comments);
     response.getWriter().println(json);
   }
 
+  /**
+   * Get the comments from the datastore based on query in GET method.
+   */
+  public List<Comment> getComments(PreparedQuery results, int limit) {
+    List<Comment> comments = new ArrayList<Comment>();
+
+    // Populate comment list until limit is reached or no comments remain.
+    int count = 0;
+    for (Entity entity : results.asIterable()) {
+      // Build the comment.
+      String text = (String) entity.getProperty(COMMENT_TEXT);
+      Date date = (Date) entity.getProperty(COMMENT_DATE);
+      Comment comment = new Comment(text, date);
+      comments.add(comment);
+
+      // Update count, and stop adding comments if limit is reached.
+      count++;
+      if (limit <= count) {
+        break;
```

Comment:
you could actually just write "return comments" here, no need for a break. That's usually stylistically preferred over breaks.

Reply:
Ah I didn't know that was preferred stylistically; I've just changed that!